ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts

```diff
@@ -141,36 +142,48 @@ export async function deleteSuccessfullyParsedMessages(
   messages: DequeuedMessageItem[],
   parseFailure: { [k: string]: boolean }
 ) {
+
+  const deletionPromises: Promise<QueueDeleteMessageResponse>[] = []
+
   for (const message of messages) {
     if (parseFailure[message.messageId]) {
       context.log(
         `Message ${message.messageId} failed to parse; skipping deletion`
       );
       continue;
     }
-    try {
-      // TODO: parallelize processing these API calls; do not await each one
-      const deleteResponse = await queueClient.deleteMessage(
-        message.messageId,
-        message.popReceipt
+
+    if(message.dequeueCount > 1){
+      context.log(
+        `Message has been dequeued ${message.dequeueCount} times, possibly sent more than once to RS`
       );
+    }
+
+    deletionPromises.push(queueClient.deleteMessage(
+      message.messageId,
+      message.popReceipt
+    ));
+  }
 
-      if(message.dequeueCount > 1){
+  try {
+    const promiseValues = await Promise.allSettled(deletionPromises);
+    for (let i = 0; i < promiseValues.length; i++) {
+      const promise = promiseValues[i];
+      const message = messages[i];
```

Comment:
if deletion is skipped, the messages and promiseValues arrays won't necessarily line up, right?

Reply:
great catch! that is correct, will fix it asap